sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala

```diff
@@ -2483,7 +2477,7 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase with Compilat
       messageParameters = Map(
         "specKeys" -> specKeys,
         "partitionColumnNames" -> partitionColumnNames.mkString(", "),
-        "tableName" -> tableName))
+        "tableName" -> toSQLId(tableName)))
```

Comment:
ditto

Reply:
If this is not modified, the following issues will occur

- When executes UT `org.apache.spark.sql.execution.command.v1.ShowTablesSuite.show table extended in multi partition key - the command's partition parameters are incomplete`, its error message is:
   ```
   org.apache.spark.sql.AnalysisException: Partition spec is invalid. The spec (id1) must match the partition spec (id1, id2) defined in table '`spark_catalog`.`ns1`.`tbl`'.
   ```
https://github.com/apache/spark/blob/efa891c4ea869a7fac58eabf187ac8aeff1fcd38/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L1343-L1346

- When executes `org.apache.spark.sql.execution.command.v2.ShowTablesSuite.show table extended in multi partition key - the command's partition parameters are incomplete`, its error message is:
   ```
   org.apache.spark.sql.AnalysisException: Partition spec is invalid. The spec (id1) must match the partition spec (id1, id2) defined in table 'test_catalog.ns1.tbl'.
   ```
https://github.com/apache/spark/blob/efa891c4ea869a7fac58eabf187ac8aeff1fcd38/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolvePartitionSpec.scala#L66



The reason is that there are two entrances to calling the method `PartitioningUtils.requireExactMatchedPartitionSpec`, 
<img width="466" alt="image" src="https://github.com/apache/spark/assets/15246973/41e3ecaa-d699-4b43-92da-53568dae3b10">

and the format of the parameter `tableName: String` inside is not consistent, so it has been unified here. In this way, we can put this UT into `show table extended in multi partition key - the command's partition parameters are incomplete` to maintain consistency.

Reply:
It seems there are two `invalidPartitionSpecError` with different signature.
Before this PR, why `org.apache.spark.sql.execution.command.v1.ShowTablesSuite.show table extended in multi partition key - the command's partition parameters are incomplete` not failed?

Reply:
Hmm... this is a newly added UT. 😄

Reply:
I got it now. In fact, you added the test cases into the base class ShowTablesSuiteBase, so the output is different from v1 and v2.
So we can add this test case into `v1.ShowTablesSuite` and `v2.ShowTablesSuite` and avoid add `toSQLId`.

After this PR, you can create another PR to fix the error class.

Reply:
Let's avoid unrelated change I said.

Reply:
@cloud-fan 
Do we really need to split a UT logic into subclasses in order not to `modify the above logic`? Separate into two subclasses, so it seems that the similarities cannot be seen anymore.

Reply:
This change is necessary as it helps to unify the v1/v2 command behavior, which is an important goal of adding new v2 commands